components/SpotForm.js

```diff
@@ -0,0 +1,169 @@
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import { formTaskOptions } from "../helpers/formTaskOptions";
+import { formEquipmentOptions } from "../helpers/formEquipmentOptions";
+export default function SpotForm({ addSpot }) {
+  const [formData, setFormData] = useState({
+    id: Date.now(),
+    Spotname: "",
+    Address: "",
+    Tasks: [],
+    Equipment: [],
+  });
+
+  const router = useRouter();
+  const handleOptionToggle = (option, field) => {
+    setFormData((prevFormData) => {
+      const selectedOptions = prevFormData[field];
+
+      if (selectedOptions.includes(option)) {
+        return {
+          ...prevFormData,
+          [field]: selectedOptions.filter(
+            (selectedOption) => selectedOption !== option
+          ),
+        };
+      } else {
+        return {
+          ...prevFormData,
+          [field]: [...selectedOptions, option],
+        };
+      }
+    });
+  };
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+
+    setFormData((prevFormData) => ({
+      ...prevFormData,
+      [name]: value,
+    }));
+  };
+
+  const onSubmitHandler = (event) => {
+    event.preventDefault();
+    const newSpot = {
+      id: formData.id,
+      Spotname: formData.Spotname, // Updated property name
+      Address: formData.Address, // Updated property name
+      Tasks: formData.Tasks, // Updated property name
+      Equipment: formData.Equipment,
+    };
+    const existingSpots = JSON.parse(localStorage.getItem("spots")) || [];
+    const updatedSpots = [...existingSpots, newSpot];
+    localStorage.setItem("spots", JSON.stringify(updatedSpots));
+    addSpot(newSpot);
+    setFormData({
+      id: Date.now(),
+      Spotname: "",
+      Address: "",
+      Tasks: [],
+      Equipment: [],
+    });
+    console.log(formData);
+    router.push("/spots");
+  };
+
+  return (
+    <FormSection>
+      <Form onSubmit={onSubmitHandler}>
+        <FormGroup>
+          <FormInputLabel htmlFor="SpotName">Spot Name:</FormInputLabel>
+          <FormInput
+            type="text"
+            name="Spotname"
+            value={formData.Spotname}
+            onChange={handleChange}
+          />
+        </FormGroup>
+        <FormGroup>
+          <FormInputLabel htmlFor="Address">Address:</FormInputLabel>
+          <FormInput
+            type="text"
+            name="Address"
+            value={formData.Address}
+            onChange={handleChange}
+          />
+        </FormGroup>
+        <FormGroup>
+          <FormInputLabel htmlFor="Tasks">Tasks:</FormInputLabel>
+          {formTaskOptions.map((option) => (
+            <FormLabel key={option}>
+              <FormInput
+                type="checkbox"
+                checked={formData.Tasks.includes(option)}
+                onChange={() => handleOptionToggle(option, "Tasks")}
+              />
+              {option}
+            </FormLabel>
+          ))}
+        </FormGroup>
```

Comment:
```suggestion
        <fieldset>
          <legend>Tasks:</legend>
          {formTaskOptions.map((option) => (
            <FormLabel key={option}>
              <FormInput
                type="checkbox"
                checked={formData.Tasks.includes(option)}
                onChange={() => handleOptionToggle(option, "Tasks")}
              />
              {option}
            </FormLabel>
          ))}
        </fieldset>
```
It is better to group your form with fieldset and legend. [See MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset)